### ISSUE TITLE
Creating a CaC enabled project results in "Branch has not been initialized"

### ISSUE THREAD

**Issue by mcasperson** [OctopusDeployLabs/terraform-provider-octopusdeploy#496]
**Describe the bug**
When you attempt to create a new CaC enabled project with the provider 0.10.5 you receive the error `Branch has not been initialized`.

The sample project used to recreate this issue is shown below:

```
terraform {
  required_providers {
    octopusdeploy = { source = "OctopusDeployLabs/octopusdeploy", version = "0.10.5" }
  }
}

provider "octopusdeploy" {
  address  = "${var.octopus_server}"
  api_key  = "${var.octopus_apikey}"
  space_id = "${var.octopus_space_id}"
}

variable "octopus_server" {
  type        = string
  nullable    = false
  sensitive   = false
  description = "The URL of the Octopus server e.g. https://myinstance.octopus.app."
}

variable "octopus_apikey" {
  type        = string
  nullable    = false
  sensitive   = true
  description = "The API key used to access the Octopus server. See https://octopus.com/docs/octopus-rest-api/how-to-create-an-api-key for details on creating an API key."
}

variable "octopus_space_id" {
  type        = string
  nullable    = false
  sensitive   = false
  description = "The ID of the Octopus space to populate."
}

data "octopusdeploy_lifecycles" "default" {
  ids          = []
  partial_name = "Default Lifecycle"
  skip         = 0
  take         = 1
}

data "octopusdeploy_worker_pools" "workerpool_hosted_ubuntu" {
  name = "Hosted Ubuntu"
  ids  = null
  skip = 0
  take = 1
}

resource "octopusdeploy_project_group" "project_group_products" {
  name        = "Products API"
  description = "The products REST API"
}


resource "octopusdeploy_git_credential" "gitcredential" {
  name     = "Octopub"
  type     = "UsernamePassword"
  username = "${var.gitusername}"
  password = "${var.gitcredential}"
}

variable "gitusername" {
  type        = string
  nullable    = false
  sensitive   = true
  description = "The git username"
}

variable "gitcredential" {
  type        = string
  nullable    = false
  sensitive   = true
  description = "The git credentials"
}

variable "giturl" {
  type        = string
  nullable    = false
  sensitive   = true
  description = "The git url"
}

variable "project_git_base_path" {
  type        = string
  nullable    = false
  sensitive   = true
  description = "The path where Config-as-Code files are saved"
  default     = "products"
}

variable "project_cac_enabled" {
  type        = string
  nullable    = false
  sensitive   = false
  description = "Enables whether the project has Config-as-Code enabled"
  default     = "true"
}

resource "octopusdeploy_project" "project_products_service" {
  auto_create_release                  = false
  default_guided_failure_mode          = "EnvironmentDefault"
  default_to_skip_if_already_installed = false
  description                          = "Deploys the backend service to Lambda."
  discrete_channel_release             = false
  included_library_variable_sets       = []
  is_disabled                          = false
  is_version_controlled                = var.project_cac_enabled
  lifecycle_id                         = "${data.octopusdeploy_lifecycles.default.lifecycles[0].id}"
  name                                 = "Products Service"
  project_group_id                     = "${octopusdeploy_project_group.project_group_products.id}"
  tenanted_deployment_participation    = "Untenanted"

  connectivity_policy {
    allow_deployments_to_no_targets = false
    exclude_unhealthy_targets       = false
    skip_machine_behavior           = "SkipUnavailableMachines"
  }

  # The connectivity policy is controlled by CaC, so this value should be ignored if
  # any settings exist and CaC is enabled
  
  lifecycle {
    ignore_changes = [
      connectivity_policy,
    ]
  }

  # This settings configure the project to use Config-as-Code

  git_library_persistence_settings {
    git_credential_id  = octopusdeploy_git_credential.gitcredential.id
    url                = var.giturl
    base_path          = ".octopus/${var.project_git_base_path}"
    default_branch     = "main"
    protected_branches = []
  }
}

resource "octopusdeploy_deployment_process" "deployment_process_project_products_service" {
  project_id = "${octopusdeploy_project.project_products_service.id}"

    step {
      condition           = "Success"
      name                = "Run a Script"
      package_requirement = "LetOctopusDecide"
      start_trigger       = "StartAfterPrevious"

      action {
        action_type                        = "Octopus.Script"
        can_be_used_for_project_versioning = false
        channels                           = []
        condition                          = "Success"
        is_disabled                        = false
        is_required                        = false
        environments                       = []
        excluded_environments              = []
        features                           = []
        name                               = "Run a Script"
        run_on_server                      = true
        tenant_tags                        = []
        worker_pool_id                     = "${data.octopusdeploy_worker_pools.workerpool_hosted_ubuntu.worker_pools[0].id}"

        properties                         = {
          "Octopus.Action.Script.Syntax" = "PowerShell"
          "Octopus.Action.Script.ScriptBody" = "echo \"hello world\""
          "Octopus.Action.Script.ScriptSource" = "Inline"
        }
    }

    properties   = {}
    target_roles = []
  }
}
```

**Steps to reproduce**
1. Save the sample HCL above to a file called `bug.tf`
2. Run `terraform init`
3. Run `terraform apply`
4. The error shown below is displayed

**Expected behavior**
The project should be created with CaC enabled.

**Logs and other supporting information**
These are the logs after running `terraform apply`:

```
$ terraform apply
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set in the CLI configuration:
│  - octopusdeploylabs/octopusdeploy in /home/matthew/Code/terraform-provider-octopusdeploy
│ 
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with
│ published releases.
╵
data.octopusdeploy_worker_pools.workerpool_hosted_ubuntu: Reading...
data.octopusdeploy_lifecycles.default: Reading...
octopusdeploy_project_group.project_group_products: Refreshing state... [id=ProjectGroups-1991]
octopusdeploy_git_credential.gitcredential: Refreshing state... [id=GitCredentials-44]
data.octopusdeploy_worker_pools.workerpool_hosted_ubuntu: Read complete after 0s [id=Worker Pools 2023-03-12 21:52:11.62775727 +0000 UTC]
data.octopusdeploy_lifecycles.default: Read complete after 0s [id=Lifecycles 2023-03-12 21:52:11.629120705 +0000 UTC]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # octopusdeploy_deployment_process.deployment_process_project_products_service will be created
  + resource "octopusdeploy_deployment_process" "deployment_process_project_products_service" {
      + branch     = (known after apply)
      + id         = (known after apply)
      + project_id = (known after apply)
      + space_id   = (known after apply)
      + version    = (known after apply)

      + step {
          + condition            = "Success"
          + condition_expression = (known after apply)
          + id                   = (known after apply)
          + name                 = "Run a Script"
          + package_requirement  = "LetOctopusDecide"
          + properties           = (known after apply)
          + start_trigger        = "StartAfterPrevious"
          + target_roles         = []

          + action {
              + action_type                        = "Octopus.Script"
              + can_be_used_for_project_versioning = false
              + channels                           = []
              + condition                          = "Success"
              + environments                       = []
              + excluded_environments              = []
              + features                           = []
              + id                                 = (known after apply)
              + is_disabled                        = false
              + is_required                        = false
              + name                               = "Run a Script"
              + properties                         = {
                  + "Octopus.Action.Script.ScriptBody"   = "echo \"hello world\""
                  + "Octopus.Action.Script.ScriptSource" = "Inline"
                  + "Octopus.Action.Script.Syntax"       = "PowerShell"
                }
              + run_on_server                      = true
              + tenant_tags                        = []
              + worker_pool_id                     = "WorkerPools-342"
            }
        }
    }

  # octopusdeploy_project.project_products_service will be created
  + resource "octopusdeploy_project" "project_products_service" {
      + auto_create_release                  = false
      + auto_deploy_release_overrides        = (known after apply)
      + cloned_from_project_id               = (known after apply)
      + default_guided_failure_mode          = "EnvironmentDefault"
      + default_to_skip_if_already_installed = false
      + deployment_changes_template          = (known after apply)
      + deployment_process_id                = (known after apply)
      + description                          = "Deploys the backend service to Lambda."
      + discrete_channel_release             = false
      + id                                   = (known after apply)
      + included_library_variable_sets       = []
      + is_disabled                          = false
      + is_discrete_channel_release          = (known after apply)
      + is_version_controlled                = true
      + lifecycle_id                         = "Lifecycles-322"
      + name                                 = "Products Service"
      + project_group_id                     = "ProjectGroups-1991"
      + release_notes_template               = (known after apply)
      + slug                                 = (known after apply)
      + space_id                             = (known after apply)
      + tenanted_deployment_participation    = "Untenanted"
      + variable_set_id                      = (known after apply)

      + connectivity_policy {
          + allow_deployments_to_no_targets = false
          + exclude_unhealthy_targets       = false
          + skip_machine_behavior           = "SkipUnavailableMachines"
          + target_roles                    = (known after apply)
        }

      + git_library_persistence_settings {
          + base_path         = (sensitive value)
          + default_branch    = "main"
          + git_credential_id = "GitCredentials-44"
          + url               = (sensitive value)
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.
╷
│ Warning: This action type has its own explicit action type
│ 
│   with octopusdeploy_deployment_process.deployment_process_project_products_service,
│   on example.tf line 149, in resource "octopusdeploy_deployment_process" "deployment_process_project_products_service":
│  149:       action_type                        = "Octopus.Script"
│ 
│ Please use the new "run_script_action" instead of the generic "action".
╵

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

octopusdeploy_project.project_products_service: Creating...
octopusdeploy_project.project_products_service: Creation complete after 2s [id=Projects-3866]
octopusdeploy_deployment_process.deployment_process_project_products_service: Creating...
╷
│ Warning: This action type has its own explicit action type
│ 
│   with octopusdeploy_deployment_process.deployment_process_project_products_service,
│   on example.tf line 149, in resource "octopusdeploy_deployment_process" "deployment_process_project_products_service":
│  149:       action_type                        = "Octopus.Script"
│ 
│ Please use the new "run_script_action" instead of the generic "action".
╵
╷
│ Error: octopus deploy api returned an error on endpoint /api/Spaces-282/projects/Projects-3866/main/deploymentprocesses - [Branch has not been initialized]
│ 
│   with octopusdeploy_deployment_process.deployment_process_project_products_service,
│   on example.tf line 139, in resource "octopusdeploy_deployment_process" "deployment_process_project_products_service":
│  139: resource "octopusdeploy_deployment_process" "deployment_process_project_products_service" {
│ 
╵
```

**Screenshots**
A branch called `octopus-vcs-conversion` is created in GitHub:

![image](https://user-images.githubusercontent.com/160104/224576043-ffcf8675-765c-49f2-99c9-24c1105c0856.png)

* `deployment_process.ocl` is empty
* `deployment_settings.ocl` is set to
```
connectivity_policy {
    skip_machine_behavior = "SkipUnavailableMachines"
}

versioning_strategy {
    template = "#{Octopus.Version.LastMajor}.#{Octopus.Version.LastMinor}.#{Octopus.Version.NextPatch}"
}
```

* `schema_version.ocl` is set to

```
version = 6
```

* `variables.ocl` is empty

**Environment and versions:**
- OS: Linux
- Octopus Server Version: v2023.2 (Build 2667)
- Terraform Version: 1.4.0
- Octopus Terraform Provider Version: 0.10.5

**Additional context**

**Comment by zentron** [comment]
An update for this one. The fix for this will include some changes to the underlying go client as well to stop forcing a dummy initial branch being used. 

There are some conceptual limitations to using Git (with protected Branches) + terraform. We cant properly model scenarios where the default branch is also protected in TF since and relying on a fake branch for creation breaks later updates.

An update is in the works

**Comment by zentron** [comment]
Fixed in v0.12.0
